addons/better-img-uploads/userscript.js

```diff
@@ -0,0 +1,203 @@
+export default async function ({ addon, console, safeMsg: m }) {
+  let mode = addon.settings.get("fitting");
+
+  addon.settings.addEventListener("change", () => {
+    mode = addon.settings.get("fitting");
+  });
+
+  //Returns html code for an item in the selection lists, complete with tooltip.
+  let html = (id, right) => `<div id="${id}">
+  <button aria-label="Upload Costume" class="${addon.tab.scratchClass("action-menu_button")} ${addon.tab.scratchClass(
+    "action-menu_more-button"
+  )} sa-better-img-uploads-btn" data-for="sa-${id}-HD Upload" data-tip="${m("upload")}" currentitem="false">
+    <img class="${addon.tab.scratchClass("action-menu_more-icon")} sa-better-img-uploader" draggable="false" src="${
+    addon.self.dir + "/icon.svg"
+  }" height="10", width="10">
+     <input accept=".svg, .png, .bmp, .jpg, .jpeg, .gif" class="${addon.tab.scratchClass(
+       "action-menu_file-input"
+     )}" multiple="" type="file">
+  </button>
+  <div class="__react_component_tooltip place-${right ? "left" : "right"} type-dark ${addon.tab.scratchClass(
+    "action-menu_tooltip"
+  )} sa-better-img-uploads-tooltip" id="sa-${id}-HD Upload" data-id="tooltip" >${m("upload")}</div>
+</div>`;
+
+  //The class name for the menu
+  let c = addon.tab.scratchClass("action-menu_more-buttons");
+
+  while (true) {
+    //Catch all upload menus as they are created
+    let menu = await addon.tab.waitForElement(`.${c}`, { markAsSeen: true });
```

Comment:
You should not use the return value of `addon.tab.scratchClass` to then run querySelector on it. It should only be used as a class name you add to elements the addon creates and owns.
You should just use an attribute selector `[class^="action-menu_more-buttons"]`

Reply:
Even with this issue the addon appears to work most of the time. But it failed here once

Reply:
Will do...